server/src/main/java/org/elasticsearch/ingest/ConfigurationUtils.java

```diff
@@ -49,6 +49,7 @@
 
     public static final String TAG_KEY = "tag";
     public static final String DESCRIPTION_KEY = "description";
+    public static final String[] VALID_MIME_TYPES = {"application/json", "text/plain", "application/x-www-form-urlencoded"};
```

Comment:
do we want the se hardcoded? What if someone specifies `application/json; charset=utf-8` ? 
Would it be of any help if this bit was unified with what CustomMustacheFactory defines? I made an attempt here https://github.com/elastic/elasticsearch/pull/66857/files#diff-207cacc1e4a5b24109a693c0b63dd762156452bb5984b5ce04499b28e6c191e8R28

Reply:
I agree that unifying this set of options would be good. I can do that after you merge that PR. I don't believe that we want to allow the specification of a particular character set for the use case of the ingest processor, though.

Reply:
Now that #66857 has been closed in favor of #67677, I believe the only improvement that can be made here is changing `content_type` to `media_type`. I'll go ahead and open a PR for that.